app/src/main/java/ru/flicksbox/network/ResponseErrorInterceptor.kt

```diff
@@ -0,0 +1,31 @@
+package ru.flicksbox.network
+
+import com.google.gson.Gson
+import okhttp3.Interceptor
+import okhttp3.ResponseBody
+import ru.flicksbox.data.ApiError
+
+class ResponseErrorInterceptor(private val gson: Gson) : Interceptor {
+    private inline fun <reified T> convertToError(body: ResponseBody): T {
+        return gson.fromJson(body.string(), T::class.java)
+    }
+
+    override fun intercept(chain: Interceptor.Chain): okhttp3.Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+        if (!response.isSuccessful) {
+            if (response.body != null) {
```

Comment:
to avoid force unwrap (`!!`) on the next line, you can use the next expression instead of `if`
```kotlin
response.body?.let { responseBody ->
    val result = convertToError<BaseDTO<Unit>>(responseBody)
    // ...
}
```